packages/@aws-cdk/aws-apigateway/lib/json-schema.ts

```diff
@@ -0,0 +1,125 @@
+export enum JsonSchemaVersion {
+  DRAFT4 = 'http://json-schema.org/draft-04/schema#',
+  DRAFT7 = 'http://json-schema.org/draft-07/schema#'
+}
+
+export enum JsonSchemaType {
+  NULL = "null",
+  BOOLEAN = "boolean",
+  OBJECT = "object",
+  ARRAY = "array",
+  NUMBER = "number",
+  INTEGER = "integer",
+  STRING = "string"
+}
+
+/**
+ * Represents a JSON schema definition of the structure of a
+ * REST API model. Copied from npm module jsonschema.
+ *
+ * @see http://json-schema.org/
+ * @see https://github.com/tdegrunt/jsonschema
+ */
+export interface JsonSchema {
+  // Special keywords
+  readonly schema?: JsonSchemaVersion;
+  readonly id?: string;
+  readonly ref?: string;
+
+  // Common properties
+  readonly type?: JsonSchemaType | JsonSchemaType[];
+  readonly title?: string;
+  readonly description?: string;
+  readonly 'enum'?: any[];
+  readonly format?: string;
+  readonly definitions?: { [name: string]: JsonSchema };
+
+  // Number or Integer
+  readonly multipleOf?: number;
+  readonly maximum?: number;
+  readonly exclusiveMaximum?: boolean;
+  readonly minimum?: number;
+  readonly exclusiveMinimum?: boolean;
+
+  // String
+  readonly maxLength?: number;
+  readonly minLength?: number;
+  readonly pattern?: string;
+
+  // Array
+  readonly items?: JsonSchema | JsonSchema[];
+  readonly additionalItems?: JsonSchema[];
+  readonly maxItems?: number;
+  readonly minItems?: number;
+  readonly uniqueItems?: boolean;
+  readonly contains?: JsonSchema | JsonSchema[];
+
+  // Object
+  readonly maxProperties?: number;
+  readonly minProperties?: number;
+  readonly required?: string[];
+  readonly properties?: { [name: string]: JsonSchema };
+  readonly additionalProperties?: JsonSchema;
+  readonly patternProperties?: { [name: string]: JsonSchema };
+  readonly dependencies?: { [name: string]: JsonSchema | string[] };
+  readonly propertyNames?: JsonSchema;
+
+  // Conditional
+  readonly allOf?: JsonSchema[];
+  readonly anyOf?: JsonSchema[];
+  readonly oneOf?: JsonSchema[];
+  readonly not?: JsonSchema;
+}
+
+export class JsonSchemaMapper {
```

Comment:
Can we make this class internal? Not sure we need it in the public API if the module

Reply:
Moved the JsonSchemaMapper in the `util.ts` file that is not exported. only left the interface and enums in `json-schema.ts` and left them exported.